dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java

```diff
@@ -570,7 +570,7 @@ public Period createPeriod(DateInterval dateInterval) {
     final DateTimeUnit from = cal.toIso(dateInterval.getFrom());
     final DateTimeUnit to = cal.toIso(dateInterval.getTo());
 
-    return new Period(this, from.toJdkDate(), to.toJdkDate(), getIsoDate(from));
+    return new Period(this, from.toJdkDate(), to.toJdkDate(), getIsoDate(dateInterval.getFrom()));
```

Comment:
Is this change intentional? I'm afraid it could break existing code relying on the old way...

Reply:
Yes, intentional. That is why the "old way" was returning Gregorian ISO names for non-gregorian calendars. See the problem description in the jira ticket.